SnowplowTracker/SnowplowTracker/Storage/EventStore.cs

```diff
@@ -0,0 +1,296 @@
+		private static readonly string DATABASE_NAME     = "URI=file:SnowplowEvents.db";
+		private static readonly string TABLE_NAME        = "events";
+		private static readonly string COLUMN_ID         = "id";
+		private static readonly string COLUMN_EVENT_DATA = "eventData";
+
+		// SQL Statements
+		private static readonly string CREATE_TABLE = "CREATE TABLE IF NOT EXISTS '" + TABLE_NAME + "' (" + COLUMN_ID + " INTEGER PRIMARY KEY, " + COLUMN_EVENT_DATA + " BLOB)";
+		private static readonly string INSERT_EVENT = "INSERT INTO " + TABLE_NAME + " (" + COLUMN_EVENT_DATA + ") " + "VALUES ";
+		private static readonly string COUNT = "SELECT COUNT(*) as events FROM " + TABLE_NAME;
+		private static readonly string SELECT_ALL = "SELECT * FROM " + TABLE_NAME;
+		private static readonly string SELECT_RANGE = "SELECT * FROM " + TABLE_NAME + " ORDER BY " + COLUMN_ID + " DESC LIMIT ";
+		private static readonly string SELECT_ROW = "SELECT * FROM " + TABLE_NAME + " WHERE " + COLUMN_ID + "=";
+		private static readonly string DELETE_ALL = "DELETE FROM " + TABLE_NAME;
+		private static readonly string DELETE_ROW = "DELETE FROM " + TABLE_NAME + " WHERE " + COLUMN_ID + "=";
+
+		// Connection
+		private IDbConnection dbConnection;
+
+		// Add Queue
+		private Thread payloadQueueConsumer;
+		private volatile bool payloadQueueConsumerContinue = true;
+		private ConcurrentQueue<TrackerPayload> payloadQueue = new ConcurrentQueue<TrackerPayload>();
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="SnowplowTracker.Storage.EventStore"/> class.
+		/// </summary>
+		public EventStore() {
+			Open ();
+			payloadQueueConsumer = new Thread(new ThreadStart(PayloadQueueConsumer));
+			payloadQueueConsumer.Start ();
+		}
+
+		/// <summary>
+		/// Opens a connection and create's the events tables if required.
+		/// </summary>
+		public void Open() {
+			if (!IsDatabaseOpen ()) {
+				Log.Verbose("EventStore: opening database: " + DATABASE_NAME);
+				dbConnection = new SqliteConnection (DATABASE_NAME);
+				dbConnection.Open();
+
+				IDbCommand command = dbConnection.CreateCommand ();
+
+				// Set database mode to Write Ahead Logging
+				command.CommandText = "PRAGMA JOURNAL_MODE = WAL;";
+				command.ExecuteNonQuery();
+
+				// Set database mode to Serialized for safe multi-threading
+				command.CommandText = "PRAGMA SQLITE_THREADSAFE = 1;";
+				command.ExecuteNonQuery();
+
+				// Create the Table if it does not exist
+				command.CommandText = CREATE_TABLE;
+				command.ExecuteNonQuery();
+
+				Log.Verbose(" + Database open status: " + IsDatabaseOpen ());
+
+				// Close command
+				command.Dispose();
+				command = null;
+			}
+		}
+
+		/// <summary>
+		/// Closes the database connection if is currently active.
+		/// </summary>
+		public void Close() {
+			if (IsDatabaseOpen ()) {
+				dbConnection.Close();
+				dbConnection.Dispose();
+				dbConnection = null;
+			}
+
+			// Close consumers
+			payloadQueueConsumerContinue = false;
+		}
```

Comment:
At the moment it doesn't look like there's a way to actually call the Close method of the EventStore instance created by the Tracker constructor. Should there be?